plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/router/MikamiTabuchiConnectionRouter.java

```diff
@@ -0,0 +1,282 @@
+    /*
+     * DBeaver - Universal Database Manager
+     * Copyright (C) 2010-2022 DBeaver Corp and others
+     *
+     * Licensed under the Apache License, Version 2.0 (the "License");
+     * you may not use this file except in compliance with the License.
+     * You may obtain a copy of the License at
+     *
+     *     http://www.apache.org/licenses/LICENSE-2.0
+     *
+     * Unless required by applicable law or agreed to in writing, software
+     * distributed under the License is distributed on an "AS IS" BASIS,
+     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+     * See the License for the specific language governing permissions and
+     * limitations under the License.
+     */
+    package org.jkiss.dbeaver.erd.ui.router;
+
+
+    import org.eclipse.draw2dl.*;
+    import org.eclipse.draw2dl.LayoutListener.Stub;
+    import org.eclipse.draw2dl.geometry.Point;
+    import org.eclipse.draw2dl.geometry.PointList;
+    import org.eclipse.draw2dl.geometry.PrecisionPoint;
+    import org.eclipse.draw2dl.geometry.Rectangle;
+    import org.jkiss.dbeaver.ui.UIUtils;
+    import org.jkiss.utils.CommonUtils;
+
+    import java.util.*;
+
+
+    public class MikamiTabuchiConnectionRouter extends AbstractRouter {
+        private boolean isDirty = false;
+
+        private final IFigure container;
+
+        private final Map<Connection, Object> constraintMap = new HashMap<>();
+        private final Map<IFigure, Rectangle> figuresToBounds = new HashMap<>();
+        private final Map<Connection, OrthogonalPath> connectionToPath = new HashMap<>();
+
+
+        private final MikamiTabuchiRouter algorithm = new MikamiTabuchiRouter();
+
+        private final Set<Connection> staleConnections = new HashSet<>();
+
+        private boolean ignoreInvalidate;
+
+        private final LayoutListener listener = new MikamiTabuchiConnectionRouter.LayoutTracker();
+        private final FigureListener figureListener = source -> {
```

Comment:
I still suggest inheriting the `MikamiTabuchiConnectionRouter` from the `FigureListener`.